groups/postgres/groups.go

```diff
@@ -165,6 +166,165 @@ func (repo groupRepository) RetrieveByID(ctx context.Context, id string) (groups
 	return toGroup(dbg)
 }
 
+func (repo groupRepository) RetrieveByIDWithRoles(ctx context.Context, id, memberID string) (groups.Group, error) {
+	query := `
+	WITH selected_group AS (
+    SELECT
+        g.id,
+        g.parent_id,
+        g.domain_id,
+        g.path AS parent_group_path
+    FROM
+        groups g
+    WHERE
+        g.id = :id
+    LIMIT 1
+	),
+	selected_group_roles AS (
+		SELECT
+			sg.id AS group_id,
+			grm.member_id AS member_id,
+			gr.id AS role_id,
+			gr.name AS role_name,
+			jsonb_agg(DISTINCT gra.action) AS actions,
+			g.path AS access_provider_path,
+			gr.entity_id AS access_provider_id,
+			CASE
+				WHEN gr.entity_id = sg.id THEN 'direct'
+				WHEN gr.entity_id = sg.parent_id THEN 'direct_group'
+				ELSE 'indirect_group'
+			END AS access_type
+		FROM
+			groups g
+		JOIN
+			groups_roles gr ON gr.entity_id = g.id
+		JOIN
+			groups_role_members grm ON gr.id = grm.role_id
+		JOIN
+			groups_role_actions gra ON gr.id = gra.role_id
+		JOIN
+			selected_group sg ON g.path @> sg.parent_group_path
+		WHERE
+			grm.member_id = :member_id
+			AND (
+				(gr.entity_id = sg.id)
+				OR (gr.entity_id = sg.parent_id AND gra.action LIKE 'group%%')
```

Comment:
Immediate parent group roles are not displaying 

Remove this condition `R (gr.entity_id = sg.parent_id AND gra.action LIKE 'group%%')` we don't have role like group%. 
```suggestion
```

Reply:
Again Immediate parent group roles are not displaying 

We should have like 
```suggestions
                OR (gr.entity_id <> sg.id AND gra.action LIKE 'subgroup%%')
```